liquid-staking/src/lib.rs

```diff
@@ -0,0 +1,66 @@
+#![no_std]
+
+use multiversx_sc::imports::*;
+
+pub mod common;
+pub mod delegation;
+pub mod delegation_proxy;
+pub mod liquid_staking_proxy;
+pub mod liquidity_pools;
+
+#[multiversx_sc::contract]
+pub trait LiquidStaking:
+    liquidity_pools::LiquidityPoolModule
+    + delegation::DelegationModule
+    + common::storage::CommonStorageModule
+{
+    #[init]
+    fn init(&self) {}
+
+    #[endpoint(registerDelegationContractAddress)]
+    fn register_delegation_address(
+        &self,
+        contract_name: ManagedBuffer,
```

Comment:
Unsure if you should allow custom names instead of assigning your internal IDs for contracts.

Reply:
I was thinking about using an enum just like in `ChainFactory` contract